Mata/Emit/MapEmit.cs

```diff
@@ -61,7 +64,12 @@ public Func<IMap<T>> Generate()
                 this.ImplementSqlDataReaderInterface();
             }
 
+#if NETSTANDARD1_6

```

Comment:
Whenever you see #if defs like this littered throughout the code, it is a code smell.  Can we make some extension methods that allow the #if defs to be isolated so that this code reads cleanly?